scripts/tools/zap/update_cluster_revisions.py

```diff
@@ -51,6 +70,34 @@ def getTargets(cluster_id: int):
     return targets
 
 
+def get_outdated_clusters(data: object, xml_clusters: dict, args) -> list[ClusterInfo]:
+    result = []
+    for endpoint in data.get("endpointTypes", []):
+        endpoint_id = endpoint.get("id") - 1
+        for cluster in endpoint.get("clusters", []):
+            # Skip cluster if user passed a cluster id and it doesn't match
+            if args.cluster_id is not None and cluster['code'] != args.cluster_id:
+                continue
+            for attribute in cluster.get("attributes", []):
+                if attribute.get("name") != "ClusterRevision" or attribute.get("storageOption") == "External":
+                    continue
+                try:
+                    cluster_revision = int(attribute.get("defaultValue"))
+                    spec_revision = xml_clusters[cluster.get("code")].revision
+                except (KeyError, ValueError):
+                    continue
+                # Filter in outdated clusters only
+                if (cluster_revision == spec_revision):
+                    break
+                # If old_revision is present, filter in matching only
+                if (args.old_revision is not None and cluster_revision != args.old_revision):
```

Comment:
Parenthesize sub-expressions:

```suggestion
                if (args.old_revision is not None) and (cluster_revision != args.old_revision):
```